morphapi/api/mpin_celldb.py

```diff
@@ -0,0 +1,140 @@
+import pandas as pd
+from pathlib import Path
+import zipfile
+import shutil
+from tqdm import tqdm
+
+
+from morphapi.paths_manager import Paths
+from morphapi.utils.data_io import connected_to_internet
+from morphapi.morphology.morphology import Neuron
+from bg_space import SpaceConvention
+from bg_atlasapi.utils import retrieve_over_http
+from bg_atlasapi import BrainGlobeAtlas
+
+
+def soma_coords_from_file(file_path):
+    """Compile dictionary with traced cells origins.
+    """
+    # Read first line after comment for soma ID:
+    line_start = "#"
+    with open(file_path, "r") as file:
+        while line_start == "#":
+            line = file.readline()
+            line_start = line[0]
+
+    return [float(p) for p in line.split(" ")[2:-2]]
+
+
+def fix_mpin_swgfile(file_path, fixed_file_path=None):
+    """Fix neurons downloaded from the MPIN website by correcting node
+    id and changing the orientation to be standard BrainGlobe.
+    """
+    if fixed_file_path is None:
+        fixed_file_path = file_path
+
+    # Fixed descriptors of the dataset space:
+    ORIGIN = "rai"
+    SHAPE = [597, 974, 359]
+    TARGET_SPACE = "asl"
+    NEW_SOMA_SIZE = 7
+
+    bgspace = SpaceConvention(origin=ORIGIN, shape=SHAPE)
+
+    df = pd.read_csv(file_path,
+                     sep=" ", header=None, comment='#')
+
+    # In this dataset, soma node is always the first, and
+    # other nodes have unspecified identity which we'll set to axon.
+    # Hopefully it will be fixed in next iterations of the database.
+    df.iloc[0, 1] = 1
+    df.iloc[1:, 1] = 2
+
+    # Map points to BrainGlobe orientation:
+    df.iloc[:, 2:-2] = bgspace.map_points_to(TARGET_SPACE, df.iloc[:, 2:-2])
+    df.iloc[0, -2] = NEW_SOMA_SIZE
+    df.to_csv(fixed_file_path, sep=" ", header=None, index=False)
+
+
+class MpinMorphology(Paths):
```

Comment:
can we rename this to MpinMorphologyAPI, for consistency with [other similar classes](https://github.com/brainglobe/morphapi/blob/master/morphapi/api/neuromorphorg.py)

Reply:
sure!